lib/xlat_tables_v2/xlat_tables_internal.c

```diff
@@ -203,9 +203,9 @@ static void xlat_tables_unmap_region(xlat_ctx_t *ctx, mmap_region_t *mm,
 				     const uintptr_t table_base_va,
 				     uint64_t *const table_base,
 				     const int table_entries,
-				     const int level)
+				     const uint level)
 {
-	assert(level >= ctx->base_level && level <= XLAT_TABLE_LEVEL_MAX);
+	assert(level >= (uint)ctx->base_level && level <= XLAT_TABLE_LEVEL_MAX);
```

Comment:
It looks like `xlat_ctx_t.ctx` should be `unsigned int` rather than casting each time it is used. Possibly other fields in that struct too.